{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "026d2f52",
   "metadata": {},
   "source": [
    "# AlphaGenome\n",
    "\n",
    "Experimenting with AlphaGenome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1de78933",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/envs/experimenting/lib/python3.10/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.27.2 is exactly one major version older than the runtime version 6.31.1 at alphagenome/protos/tensor.proto. Please update the gencode to avoid compatibility violations in the next runtime release.\n",
      "  warnings.warn(\n",
      "/opt/conda/envs/experimenting/lib/python3.10/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.27.2 is exactly one major version older than the runtime version 6.31.1 at alphagenome/protos/dna_model.proto. Please update the gencode to avoid compatibility violations in the next runtime release.\n",
      "  warnings.warn(\n",
      "/opt/conda/envs/experimenting/lib/python3.10/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n",
      "/opt/conda/envs/experimenting/lib/python3.10/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    }
   ],
   "source": [
    "from alphagenome.models.variant_scorers import GeneMaskLFCScorer\n",
    "from alphagenome.models.dna_client import OutputType\n",
    "from alphagenome.data import genome\n",
    "from alphagenome.models import variant_scorers, dna_client"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9e71ebe1",
   "metadata": {},
   "outputs": [],
   "source": [
    "alphagenome_api_key = \"AIzaSyCir34AonB_Q0y7aHJx1od6Z5SJrf5Z1mg\"\n",
    "model = dna_client.create(alphagenome_api_key)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "58079867",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>scored_interval</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_strand</th>\n",
       "      <th>junction_Start</th>\n",
       "      <th>junction_End</th>\n",
       "      <th>output_type</th>\n",
       "      <th>variant_scorer</th>\n",
       "      <th>track_name</th>\n",
       "      <th>track_strand</th>\n",
       "      <th>Assay title</th>\n",
       "      <th>ontology_curie</th>\n",
       "      <th>biosample_name</th>\n",
       "      <th>biosample_type</th>\n",
       "      <th>gtex_tissue</th>\n",
       "      <th>raw_score</th>\n",
       "      <th>quantile_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2:25234373:C&gt;T</td>\n",
       "      <td>chr2:25168837-25299909:.</td>\n",
       "      <td>ENSG00000115138</td>\n",
       "      <td>POMC</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>CL:0000047 polyA plus RNA-seq</td>\n",
       "      <td>-</td>\n",
       "      <td>polyA plus RNA-seq</td>\n",
       "      <td>CL:0000047</td>\n",
       "      <td>neuronal stem cell</td>\n",
       "      <td>in_vitro_differentiated_cells</td>\n",
       "      <td></td>\n",
       "      <td>0.000042</td>\n",
       "      <td>0.160114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2:25234373:C&gt;T</td>\n",
       "      <td>chr2:25168837-25299909:.</td>\n",
       "      <td>ENSG00000115138</td>\n",
       "      <td>POMC</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>CL:0000062 total RNA-seq</td>\n",
       "      <td>-</td>\n",
       "      <td>total RNA-seq</td>\n",
       "      <td>CL:0000062</td>\n",
       "      <td>osteoblast</td>\n",
       "      <td>primary_cell</td>\n",
       "      <td></td>\n",
       "      <td>0.000001</td>\n",
       "      <td>0.034594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2:25234373:C&gt;T</td>\n",
       "      <td>chr2:25168837-25299909:.</td>\n",
       "      <td>ENSG00000115138</td>\n",
       "      <td>POMC</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>CL:0000084 polyA plus RNA-seq</td>\n",
       "      <td>-</td>\n",
       "      <td>polyA plus RNA-seq</td>\n",
       "      <td>CL:0000084</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>primary_cell</td>\n",
       "      <td></td>\n",
       "      <td>-0.000015</td>\n",
       "      <td>-0.092027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr2:25234373:C&gt;T</td>\n",
       "      <td>chr2:25168837-25299909:.</td>\n",
       "      <td>ENSG00000115138</td>\n",
       "      <td>POMC</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>CL:0000084 total RNA-seq</td>\n",
       "      <td>-</td>\n",
       "      <td>total RNA-seq</td>\n",
       "      <td>CL:0000084</td>\n",
       "      <td>T-cell</td>\n",
       "      <td>primary_cell</td>\n",
       "      <td></td>\n",
       "      <td>-0.000005</td>\n",
       "      <td>-0.046111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr2:25234373:C&gt;T</td>\n",
       "      <td>chr2:25168837-25299909:.</td>\n",
       "      <td>ENSG00000115138</td>\n",
       "      <td>POMC</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>CL:0000115 total RNA-seq</td>\n",
       "      <td>-</td>\n",
       "      <td>total RNA-seq</td>\n",
       "      <td>CL:0000115</td>\n",
       "      <td>endothelial cell</td>\n",
       "      <td>in_vitro_differentiated_cells</td>\n",
       "      <td></td>\n",
       "      <td>0.000056</td>\n",
       "      <td>0.160114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1183</th>\n",
       "      <td>chr2:25234373:C&gt;T</td>\n",
       "      <td>chr2:25168837-25299909:.</td>\n",
       "      <td>ENSG00000230452</td>\n",
       "      <td>LINC01381</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>UBERON:0018115 polyA plus RNA-seq</td>\n",
       "      <td>.</td>\n",
       "      <td>polyA plus RNA-seq</td>\n",
       "      <td>UBERON:0018115</td>\n",
       "      <td>left renal pelvis</td>\n",
       "      <td>tissue</td>\n",
       "      <td></td>\n",
       "      <td>-0.000320</td>\n",
       "      <td>-0.322598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1184</th>\n",
       "      <td>chr2:25234373:C&gt;T</td>\n",
       "      <td>chr2:25168837-25299909:.</td>\n",
       "      <td>ENSG00000230452</td>\n",
       "      <td>LINC01381</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>UBERON:0018116 polyA plus RNA-seq</td>\n",
       "      <td>.</td>\n",
       "      <td>polyA plus RNA-seq</td>\n",
       "      <td>UBERON:0018116</td>\n",
       "      <td>right renal pelvis</td>\n",
       "      <td>tissue</td>\n",
       "      <td></td>\n",
       "      <td>0.000154</td>\n",
       "      <td>0.215740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1185</th>\n",
       "      <td>chr2:25234373:C&gt;T</td>\n",
       "      <td>chr2:25168837-25299909:.</td>\n",
       "      <td>ENSG00000230452</td>\n",
       "      <td>LINC01381</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>UBERON:0018117 polyA plus RNA-seq</td>\n",
       "      <td>.</td>\n",
       "      <td>polyA plus RNA-seq</td>\n",
       "      <td>UBERON:0018117</td>\n",
       "      <td>left renal cortex interstitium</td>\n",
       "      <td>tissue</td>\n",
       "      <td></td>\n",
       "      <td>0.000219</td>\n",
       "      <td>0.248480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1186</th>\n",
       "      <td>chr2:25234373:C&gt;T</td>\n",
       "      <td>chr2:25168837-25299909:.</td>\n",
       "      <td>ENSG00000230452</td>\n",
       "      <td>LINC01381</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>UBERON:0018118 polyA plus RNA-seq</td>\n",
       "      <td>.</td>\n",
       "      <td>polyA plus RNA-seq</td>\n",
       "      <td>UBERON:0018118</td>\n",
       "      <td>right renal cortex interstitium</td>\n",
       "      <td>tissue</td>\n",
       "      <td></td>\n",
       "      <td>-0.000208</td>\n",
       "      <td>-0.248480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1187</th>\n",
       "      <td>chr2:25234373:C&gt;T</td>\n",
       "      <td>chr2:25168837-25299909:.</td>\n",
       "      <td>ENSG00000230452</td>\n",
       "      <td>LINC01381</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>RNA_SEQ</td>\n",
       "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
       "      <td>UBERON:0036149 gtex Skin_Not_Sun_Exposed_Supra...</td>\n",
       "      <td>.</td>\n",
       "      <td>polyA plus RNA-seq</td>\n",
       "      <td>UBERON:0036149</td>\n",
       "      <td>suprapubic skin</td>\n",
       "      <td>tissue</td>\n",
       "      <td>Skin_Not_Sun_Exposed_Suprapubic</td>\n",
       "      <td>0.000229</td>\n",
       "      <td>0.204714</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1188 rows Ã— 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             variant_id           scored_interval          gene_id  gene_name  \\\n",
       "0     chr2:25234373:C>T  chr2:25168837-25299909:.  ENSG00000115138       POMC   \n",
       "1     chr2:25234373:C>T  chr2:25168837-25299909:.  ENSG00000115138       POMC   \n",
       "2     chr2:25234373:C>T  chr2:25168837-25299909:.  ENSG00000115138       POMC   \n",
       "3     chr2:25234373:C>T  chr2:25168837-25299909:.  ENSG00000115138       POMC   \n",
       "4     chr2:25234373:C>T  chr2:25168837-25299909:.  ENSG00000115138       POMC   \n",
       "...                 ...                       ...              ...        ...   \n",
       "1183  chr2:25234373:C>T  chr2:25168837-25299909:.  ENSG00000230452  LINC01381   \n",
       "1184  chr2:25234373:C>T  chr2:25168837-25299909:.  ENSG00000230452  LINC01381   \n",
       "1185  chr2:25234373:C>T  chr2:25168837-25299909:.  ENSG00000230452  LINC01381   \n",
       "1186  chr2:25234373:C>T  chr2:25168837-25299909:.  ENSG00000230452  LINC01381   \n",
       "1187  chr2:25234373:C>T  chr2:25168837-25299909:.  ENSG00000230452  LINC01381   \n",
       "\n",
       "           gene_type gene_strand junction_Start junction_End output_type  \\\n",
       "0     protein_coding           -           None         None     RNA_SEQ   \n",
       "1     protein_coding           -           None         None     RNA_SEQ   \n",
       "2     protein_coding           -           None         None     RNA_SEQ   \n",
       "3     protein_coding           -           None         None     RNA_SEQ   \n",
       "4     protein_coding           -           None         None     RNA_SEQ   \n",
       "...              ...         ...            ...          ...         ...   \n",
       "1183          lncRNA           +           None         None     RNA_SEQ   \n",
       "1184          lncRNA           +           None         None     RNA_SEQ   \n",
       "1185          lncRNA           +           None         None     RNA_SEQ   \n",
       "1186          lncRNA           +           None         None     RNA_SEQ   \n",
       "1187          lncRNA           +           None         None     RNA_SEQ   \n",
       "\n",
       "                                   variant_scorer  \\\n",
       "0     GeneMaskLFCScorer(requested_output=RNA_SEQ)   \n",
       "1     GeneMaskLFCScorer(requested_output=RNA_SEQ)   \n",
       "2     GeneMaskLFCScorer(requested_output=RNA_SEQ)   \n",
       "3     GeneMaskLFCScorer(requested_output=RNA_SEQ)   \n",
       "4     GeneMaskLFCScorer(requested_output=RNA_SEQ)   \n",
       "...                                           ...   \n",
       "1183  GeneMaskLFCScorer(requested_output=RNA_SEQ)   \n",
       "1184  GeneMaskLFCScorer(requested_output=RNA_SEQ)   \n",
       "1185  GeneMaskLFCScorer(requested_output=RNA_SEQ)   \n",
       "1186  GeneMaskLFCScorer(requested_output=RNA_SEQ)   \n",
       "1187  GeneMaskLFCScorer(requested_output=RNA_SEQ)   \n",
       "\n",
       "                                             track_name track_strand  \\\n",
       "0                         CL:0000047 polyA plus RNA-seq            -   \n",
       "1                              CL:0000062 total RNA-seq            -   \n",
       "2                         CL:0000084 polyA plus RNA-seq            -   \n",
       "3                              CL:0000084 total RNA-seq            -   \n",
       "4                              CL:0000115 total RNA-seq            -   \n",
       "...                                                 ...          ...   \n",
       "1183                  UBERON:0018115 polyA plus RNA-seq            .   \n",
       "1184                  UBERON:0018116 polyA plus RNA-seq            .   \n",
       "1185                  UBERON:0018117 polyA plus RNA-seq            .   \n",
       "1186                  UBERON:0018118 polyA plus RNA-seq            .   \n",
       "1187  UBERON:0036149 gtex Skin_Not_Sun_Exposed_Supra...            .   \n",
       "\n",
       "             Assay title  ontology_curie                   biosample_name  \\\n",
       "0     polyA plus RNA-seq      CL:0000047               neuronal stem cell   \n",
       "1          total RNA-seq      CL:0000062                       osteoblast   \n",
       "2     polyA plus RNA-seq      CL:0000084                           T-cell   \n",
       "3          total RNA-seq      CL:0000084                           T-cell   \n",
       "4          total RNA-seq      CL:0000115                 endothelial cell   \n",
       "...                  ...             ...                              ...   \n",
       "1183  polyA plus RNA-seq  UBERON:0018115                left renal pelvis   \n",
       "1184  polyA plus RNA-seq  UBERON:0018116               right renal pelvis   \n",
       "1185  polyA plus RNA-seq  UBERON:0018117   left renal cortex interstitium   \n",
       "1186  polyA plus RNA-seq  UBERON:0018118  right renal cortex interstitium   \n",
       "1187  polyA plus RNA-seq  UBERON:0036149                  suprapubic skin   \n",
       "\n",
       "                     biosample_type                      gtex_tissue  \\\n",
       "0     in_vitro_differentiated_cells                                    \n",
       "1                      primary_cell                                    \n",
       "2                      primary_cell                                    \n",
       "3                      primary_cell                                    \n",
       "4     in_vitro_differentiated_cells                                    \n",
       "...                             ...                              ...   \n",
       "1183                         tissue                                    \n",
       "1184                         tissue                                    \n",
       "1185                         tissue                                    \n",
       "1186                         tissue                                    \n",
       "1187                         tissue  Skin_Not_Sun_Exposed_Suprapubic   \n",
       "\n",
       "      raw_score  quantile_score  \n",
       "0      0.000042        0.160114  \n",
       "1      0.000001        0.034594  \n",
       "2     -0.000015       -0.092027  \n",
       "3     -0.000005       -0.046111  \n",
       "4      0.000056        0.160114  \n",
       "...         ...             ...  \n",
       "1183  -0.000320       -0.322598  \n",
       "1184   0.000154        0.215740  \n",
       "1185   0.000219        0.248480  \n",
       "1186  -0.000208       -0.248480  \n",
       "1187   0.000229        0.204714  \n",
       "\n",
       "[1188 rows x 19 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Define the DNMT3A R882C variant\n",
    "variant = genome.Variant(\n",
    "    chromosome=\"chr2\", position=25234373, reference_bases=\"C\", alternate_bases=\"T\"\n",
    ")\n",
    "\n",
    "# Create interval around the variant\n",
    "sequence_length = dna_client.SUPPORTED_SEQUENCE_LENGTHS[\"SEQUENCE_LENGTH_100KB\"]\n",
    "interval = variant.reference_interval.resize(sequence_length)\n",
    "\n",
    "# Create the RNA-seq LFC scorer\n",
    "rna_lfc_scorer = GeneMaskLFCScorer(requested_output=OutputType.RNA_SEQ)\n",
    "\n",
    "# Score the variant for RNA-seq\n",
    "rna_scores = model.score_variant(\n",
    "    interval=interval, variant=variant, variant_scorers=[rna_lfc_scorer]\n",
    ")\n",
    "\n",
    "# Convert to dataframe\n",
    "rna_df = variant_scorers.tidy_scores(rna_scores)\n",
    "rna_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b344628c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of predictions: 1188\n",
      "Mean LFC: -0.001372\n",
      "Median LFC: -0.000035\n",
      "Min LFC: -0.024788\n",
      "Max LFC: 0.011298\n",
      "Upregulated: 409\n",
      "Downregulated: 751\n",
      "Top upregulated:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biosample_name</th>\n",
       "      <th>raw_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>827</th>\n",
       "      <td>mononuclear cell</td>\n",
       "      <td>0.011298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>826</th>\n",
       "      <td>hematopoietic multipotent progenitor cell</td>\n",
       "      <td>0.010371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>530</th>\n",
       "      <td>HeLa-S3</td>\n",
       "      <td>0.009999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1075</th>\n",
       "      <td>immature natural killer cell</td>\n",
       "      <td>0.007974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>809</th>\n",
       "      <td>B cell</td>\n",
       "      <td>0.007563</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 biosample_name  raw_score\n",
       "827                            mononuclear cell   0.011298\n",
       "826   hematopoietic multipotent progenitor cell   0.010371\n",
       "530                                     HeLa-S3   0.009999\n",
       "1075               immature natural killer cell   0.007974\n",
       "809                                      B cell   0.007563"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Top downregulated:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biosample_name</th>\n",
       "      <th>raw_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>668</th>\n",
       "      <td>Purkinje cell</td>\n",
       "      <td>-0.024788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>708</th>\n",
       "      <td>GM23248</td>\n",
       "      <td>-0.016414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>781</th>\n",
       "      <td>dorsolateral prefrontal cortex</td>\n",
       "      <td>-0.014448</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500</th>\n",
       "      <td>hair follicular keratinocyte</td>\n",
       "      <td>-0.014296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510</th>\n",
       "      <td>MCF 10A</td>\n",
       "      <td>-0.013721</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     biosample_name  raw_score\n",
       "668                   Purkinje cell  -0.024788\n",
       "708                         GM23248  -0.016414\n",
       "781  dorsolateral prefrontal cortex  -0.014448\n",
       "500    hair follicular keratinocyte  -0.014296\n",
       "510                         MCF 10A  -0.013721"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "print(f\"Number of predictions: {len(rna_df)}\")\n",
    "\n",
    "# Summary statistics\n",
    "print(f\"Mean LFC: {rna_df['raw_score'].mean():.6f}\")\n",
    "print(f\"Median LFC: {rna_df['raw_score'].median():.6f}\")\n",
    "print(f\"Min LFC: {rna_df['raw_score'].min():.6f}\")\n",
    "print(f\"Max LFC: {rna_df['raw_score'].max():.6f}\")\n",
    "\n",
    "# Count effects\n",
    "upregulated = (rna_df[\"raw_score\"] > 0).sum()\n",
    "downregulated = (rna_df[\"raw_score\"] < 0).sum()\n",
    "print(f\"Upregulated: {upregulated}\")\n",
    "print(f\"Downregulated: {downregulated}\")\n",
    "\n",
    "# Top effects\n",
    "top_up = rna_df.nlargest(5, \"raw_score\")[[\"biosample_name\", \"raw_score\"]]\n",
    "top_down = rna_df.nsmallest(5, \"raw_score\")[[\"biosample_name\", \"raw_score\"]]\n",
    "\n",
    "print(\"Top upregulated:\")\n",
    "display(top_up)\n",
    "print(\"Top downregulated:\")\n",
    "display(top_down)\n",
    "\n",
    "rna_df.to_csv(\"DNMT3A_R882C_RNA_seq_LFC_scores.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7d02b121",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "experimenting",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
